### PR TITLE
[PR #6843/f90cb195 backport][3.9] Add type Path on web app run

### PR DESCRIPTION
**Pull request by SamirPS**
**This is a backport of PR #6843 as merged into master (f90cb1952c272936845b93a8f424b32e703d545c).**

## What do these changes do?

I modify the web.py file and add Union[str,pathlib.Path] for path

## Are there changes in behavior for the user?

For the option of path, we can have str or pathlib.path

## Related issue number

#6839 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is &lt;Name&gt; &lt;Surname&gt;.
  * Please keep alphabetical order, the file is sorted by names.
- [x] Add a new news fragment into the `CHANGES` folder
  * name it `<issue_id>.<type>` for example (588.bugfix)
  * if you don't have an `issue_id` change it to the pr id after creating the pr
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example: "Fix issue with non-ascii contents in doctest text files."

### CODE REVIEW

**Comment by SamirPS** [comment]
@Dreamsorcerer it's good now ? i follow all the step

**Comment by webknjaz** [comment]
You used the wrong target branch. The PR shouldn't be filed against `master`.

**Comment by SamirPS** [comment]
i change the base to 3.9, it's good ?

**Comment by SamirPS** [comment]
The CI / Test (3.7, windows, false) (pull_request) has failed because of 
```
Requirement already satisfied: pyparsing<3,>=2.0.2 in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (from packaging->pytest==6.2.5->-r requirements\lint.txt (line 5)) (2.4.7)
ERROR: To modify pip, please run the following command:
C:\hostedtoolcache\windows\Python\3.7.9\x64\python.exe -m pip install -r requirements/dev.txt -c requirements/constraints.txt
mingw32-make: *** [makefile:77: .install-deps] Error 1
Error: Process completed with exit code 1.

````

What can i do ? because i don't thinks it's come from me

**Comment by webknjaz** [comment]
> i change the base to 3.9, it's good ?

It should be.

**Comment by SamirPS** [comment]
> > i change the base to 3.9, it's good ?
> 
> It should be.

Thanks, i already do it, now i have problem with some test 

https://github.com/aio-libs/aiohttp/pull/6874#issuecomment-1208190356

File: aiohttp/web_runner.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 import socket
 from abc import ABC, abstractmethod
 from pathlib import Path
-from typing import Any, List, Optional, Set, Type, Union
```
Comment (webknjaz) [inline aiohttp/web_runner.py]: Is this change absolutely necessary? Why wasn't it included in the original PR, then? Usually, backports should be as close as possible to the original PRs. If this is _also_ necessary on master, it should be first submitted there and backported separately.

**Comment by webknjaz** [comment]
> The CI / Test (3.7, windows, false) (pull_request) has failed because of
> 
> ```
> Requirement already satisfied: pyparsing<3,>=2.0.2 in c:\hostedtoolcache\windows\python\3.7.9\x64\lib\site-packages (from packaging->pytest==6.2.5->-r requirements\lint.txt (line 5)) (2.4.7)
> ERROR: To modify pip, please run the following command:
> C:\hostedtoolcache\windows\Python\3.7.9\x64\python.exe -m pip install -r requirements/dev.txt -c requirements/constraints.txt
> mingw32-make: *** [makefile:77: .install-deps] Error 1
> Error: Process completed with exit code 1.
> ```
> 
> What can i do ? because i don't thinks it's come from me @webknjaz @Dreamsorcerer

You're right. It's most likely my fault. The repository is not in a clean state right now, as we're trying to make compatibility changes for the Python 3.11 support.
So until we fix the CI+requirements setup in the branch, we cannot test and merge this one. It'll have to wait until we sort it out, and then rebased to restart the CI with the respective updates to the branch.

File: aiohttp/web_runner.py (hunk 1 above)
Comment (SamirPS) [reply aiohttp/web_runner.py]: i had this on before the backported . I just remove the Type because LGTM tell me it's not use

**Review by Dreamsorcerer — APPROVED** [review]

**Comment by webknjaz** [comment]
@SamirPS could you rebase this resolving the conflicts?

**Comment by webknjaz** [comment]
Looks like the fork has been deleted along with the branch. I'll have to re-submit this.

**Comment by webknjaz** [comment]
Superseded by https://github.com/aio-libs/aiohttp/pull/6957.

**Pull request closed without merging**